# Why Learn to code

 Software and technology are becoming increasingly integrated into our
everyday lives, allowing us to accomplish tasks, navigate to destinations,
make purchases, and stay connected with friends.


 While some may choose to study computer science as part
of their formal education, everyone can benefit from an understanding of
algorithmic thinking and computational processes. Learning how the
software that we use on a daily basis is made can allow us as end users to
evaluate how and why these applications are developed, enabling us to
think critically about these tools and how to improve them.

 While not everyone needs to learn to code
complex programs, learning how coding works can help shape the future
of technology and increase the number of stakeholders, decision makers,
and knowledge producers who can work to build better software for
everyone.

  Computer programming provides many applications across
domains, and can help us solve problems in specialities such as medicine,
economics, sociology, history, and literature, to name a few. By
integrating technology’s methodologies into our own fields, we can
leverage computational logic and software design and development
practices in our work.

# Why Learn Python

Extremely versatile and popular among developers, Python is a good
general-purpose language that can be used in a variety of applications.
For those with an understanding of English, Python is a very humanreadable
programming language, allowing for quick comprehension.

Because Python supports multiple styles including scripting and objectoriented
programming, it is considered to be a multi-paradigm language
that enables programmers to use the most suitable style to complete a
project.

Increasingly used in industry, Python offers a lot of potential for
those who would like to begin coding while also being a good choice for
those looking to pick up an additional programming language.

# Python Installation and Develepment Enviroment

Python Installation:you can download and install python in your favorite os from official python website;python.org
Python Ide's & Text Editors..
Pycharm,Sublime,VScode,Atom...My favorite juypter notebooks.
You can install python from anaconda distribution

# Python 2 vs Python 3 Practical considerations

Python 2
Published in late 2000, Python 2 signalled a more transparent and
inclusive language development process than earlier versions of Python
with the implementation of PEP (Python Enhancement Proposal), a
technical specification that either provides information to Python
community members or describes a new feature of the language.

Additionally, Python 2 included many more programmatic features
including a cycle-detecting garbage collector to automate memory
management, increased Unicode support to standardize characters, and
list comprehensions to create a list based on existing lists. As Python 2
continued to develop, more features were added, including unifying
Python’s types and classes into one hierarchy in Python version 2.2.

Python 3 is regarded as the future of Python and is the version of the
language that is currently in development. A major overhaul, Python 3
was released in late 2008 to address and amend intrinsic design flaws of
previous versions of the language. The focus of Python 3 development
was to clean up the codebase and remove redundancy, making it clear
that there was only one way to perform a given task.

  Major modifications to Python 3.0 included changing the print
statement into a built-in function, improve the way integers are divided,
and providing more Unicode support.

### Key Differences

Print   
Python 2 print is treated as a statment but in Python 3 print is treated as function.

In [6]:
#print in python 2 

#print"My name is Prosperity"
#print in python 3

print('My name is Prosperity')


My name is Prosperity


  Division with Integers
  
  In Python 3, integer division became more intuitive, as in:


In [7]:
a = 5 / 2
print(a)

2.5


In Python 2, any number that you type without decimals is treated as the
programming type called integer. While at first glance this seems like an
easy way to handle programming types, when you try to divide integers
together sometimes you expect to get an answer with decimal places
(called a float), as in:

5 / 2 = 2.5

However, in Python 2 integers were strongly typed and would not
change to a float with decimal places even in cases when that would
make intuitive sense.

When the two numbers on either side of the division / symbol are
integers, Python 2 does floor division so that for the quotient x the
number returned is the largest integer less than or equal to x.

This means
that when you write 5 / 2 
to divide the two numbers, Python 2.7
returns the largest integer less than or equal to 2.5, in this case 2:

a = 5 / 2
print a   
output 2